### PR TITLE
[wasm] Add a try-catch around GetCallingAssembly () calls in ResourceManager, it can fail on wasm aot.

### PR DESCRIPTION
**Pull request by vargaz**
Fixes https://github.com/mono/mono/issues/19314.

### CODE REVIEW

**Comment by vargaz** [comment]
@monojenkins build failed

File: mcs/class/referencesource/mscorlib/system/resources/resourcemanager.cs [hunk 1]
```diff
@@ -372,15 +375,17 @@ public ResourceManager(String baseName, Assembly assembly)
...
+			try {
+				m_callingAssembly = (RuntimeAssembly)Assembly.GetCallingAssembly();
+				// Special case for mscorlib - protect mscorlib's private resources.
+				// This isn't for security reasons, but to ensure we can make
+				// breaking changes to mscorlib's internal resources without 
+				// assuming users may have taken a dependency on them.
+				if (assembly == typeof(Object).Assembly && m_callingAssembly != assembly)
+					m_callingAssembly = null;
+			} catch {
+			}
```
Comment (kjpou1) [inline mcs/class/referencesource/mscorlib/system/resources/resourcemanager.cs]: Indentation looks off.

File: mcs/class/referencesource/mscorlib/system/resources/resourcemanager.cs [hunk 2]
```diff
@@ -437,12 +445,13 @@ public ResourceManager(Type resourceSource)
...
-            {
-                m_callingAssembly = null;
-            }
+			try {
+				m_callingAssembly = (RuntimeAssembly)Assembly.GetCallingAssembly();
+				// Special case for mscorlib - protect mscorlib's private resources.
+				if (MainAssembly == typeof(Object).Assembly && m_callingAssembly != MainAssembly)
+					m_callingAssembly = null;
+            } catch {
+			}
```
Comment (kjpou1) [inline mcs/class/referencesource/mscorlib/system/resources/resourcemanager.cs]: Here as well

File: mcs/class/referencesource/mscorlib/system/resources/resourcemanager.cs [hunk 3]
```diff
@@ -410,13 +415,16 @@ public ResourceManager(String baseName, Assembly assembly, Type usingResourceSet
...
+			try {
+				m_callingAssembly = (RuntimeAssembly)Assembly.GetCallingAssembly();
+				// Special case for mscorlib - protect mscorlib's private resources.
+				// This isn't for security reasons, but to ensure we can make
+				// breaking changes to mscorlib's internal resources without 
+				// assuming users may have taken a dependency on them.
+				if (assembly == typeof(Object).Assembly && m_callingAssembly != assembly)
+					m_callingAssembly = null;
+			} catch {
+			}
```
Comment (kjpou1) [inline mcs/class/referencesource/mscorlib/system/resources/resourcemanager.cs]: Here as well

**Comment by vargaz** [comment]
@monojenkins build failed

File: mcs/class/referencesource/mscorlib/system/resources/resourcemanager.cs [hunk 4]
```diff
@@ -296,7 +299,7 @@ protected ResourceManager()
             ResourceManagerMediator mediator = new ResourceManagerMediator(this);
             resourceGroveler = new ManifestBasedResourceGroveler(mediator);
         }
-	
+
```
Comment (steveisok) [inline mcs/class/referencesource/mscorlib/system/resources/resourcemanager.cs]: Nit: unneeded space

File: mcs/class/referencesource/mscorlib/system/resources/resourcemanager.cs [hunk 5]
```diff
@@ -372,14 +375,16 @@ public ResourceManager(String baseName, Assembly assembly)
...
+            // GetCallingAssembly might not work on some platforms
+            try {
+                m_callingAssembly = (RuntimeAssembly)Assembly.GetCallingAssembly();
+                // Special case for mscorlib - protect mscorlib's private resources.
+                // This isn't for security reasons, but to ensure we can make
+                // breaking changes to mscorlib's internal resources without 
+                // assuming users may have taken a dependency on them.
+                if (assembly == typeof(Object).Assembly && m_callingAssembly != assembly)
+                    m_callingAssembly = null;
+            } catch {
```
Comment (steveisok) [inline mcs/class/referencesource/mscorlib/system/resources/resourcemanager.cs]: Do we want the empty catch to apply to all profiles?  I don't have a sense of the implication.

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**